### PR TITLE
🎉 New Source: Mailjet Mail API

### PR DESCRIPTION
**Pull request by haithem-souala**
## What
https://dev.mailjet.com/email

## How
A new source connector using low-code CDK

## Streams
1. ContactsList
2. Contacts
3. Stats
4. Message
5. Campaigns

## 🚨 User Impact 🚨
N/A

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*
Capture d’écran 2022-10-23 à 22 09 23

</details>

### CODE REVIEW

**Comment by haithem-souala** [comment]
Capture d’écran 2022-10-23 à 22 09 23

File: .vscode/frontend.code-workspace [hunk 1]
```diff
@@ -1,71 +0,0 @@
-{
-  "folders": [
-    {
-      "path": "../airbyte-webapp"
```
Comment (marcosmarxm) [inline .vscode/frontend.code-workspace:4]: Please check this commit and restore these two files.

File: airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+connectionSpecification:
+  $schema: http://json-schema.org/draft-07/schema#
+  title: Mailjet Mail Spec
+  type: object
+  required:
+    - username
+    - password
+  additionalProperties: true
+  properties:
+    username:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/spec.yaml:11]:
```suggestion
    api_key:
```
would be better call `api_key` and `api_key_secret`?

File: airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/schemas/TODO.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# TODO: Define your stream schemas
+Your connector must describe the schema of each stream it can output using [JSONSchema](https://json-schema.org). 
+
+You can describe the schema of your streams using one `.json` file per stream.
+
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/schemas/TODO.md:5]: Remove this file.

File: airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/mailjet_mail.yaml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    $options:
+      name: "stats"
+      primary_key: "APIKeyID"
+      path: "/statcounters?CounterSource=APIKey&CounterResolution=Lifetime&CounterTiming=Message"
+  campaign_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "campaign"
+      primary_key: "ID"
+      path: "/campaign?FromTS=972314998"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/mailjet_mail.yaml:67]: There is any documentation saying why to use this FromTS?

File: airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/mailjet_mail.yaml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  contactslist_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "contactslist"
+      primary_key: "ID"
+      path: "/contactslist?limit=100"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/mailjet_mail.yaml:42]: Why this limit here?

File: airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/mailjet_mail.yaml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    $options:
+      name: "contacts"
+      primary_key: "ID"
+      path: "/contact"
+  stats_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "stats"
+      primary_key: "APIKeyID"
+      path: "/statcounters?CounterSource=APIKey&CounterResolution=Lifetime&CounterTiming=Message"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/mailjet_mail.yaml:61]: This is to request a specific stats?

File: docs/integrations/sources/mailjet-mail.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+### Performance considerations
+
+Mailjet APIs are under rate limits for the number of API calls allowed per API keys per second. If you reach a rate limit, API will return a 429 HTTP error code. See [here](https://dev.mailjet.com/email/reference/overview/rate-limits/)
+
+## Getting started
+
+### Requirements
+
+* Mailjet Mail API_KEY (username)
+* Mailjet Mail SECRET_KEY (password)
```
Comment (marcosmarxm) [inline docs/integrations/sources/mailjet-mail.md:31]: Please add the changelog part here. Check other connector docs.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments, look good!

File: airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/mailjet_mail.yaml (hunk 4 above)
Comment (haithem-souala) [reply airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/mailjet_mail.yaml]: Indeed, it was a test, my bad.

File: airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/mailjet_mail.yaml (hunk 6 above)
Comment (haithem-souala) [reply airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/mailjet_mail.yaml:68]: I changed the stream name, and yes it's a specific stats (Mailjet API design is weird :) )

**Comment by haithem-souala** [comment]
Capture d’écran 2022-10-25 à 11 04 19

File: airbyte-integrations/connectors/source-mailjet-mail/integration_tests/sample_config.json [hunk 8]
```diff
@@ -0,0 +1,3 @@
+{
+    "fix-me": "TODO"
+}
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-mail/integration_tests/sample_config.json:3]: Please update this file and give a config example.

File: airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/spec.yaml [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  required:
+    - api_key
+    - api_key_secret
+  additionalProperties: true
+  properties:
+    api_key:
+      type: string
+      description: >-
+        Your API Key. See <a
+        href="https://dev.mailjet.com/email/guides/#authentication">here</a>.
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/spec.yaml:15]:
```suggestion
    api_key:
      title: API Key
      type: string
      description: >-
        Your API Key. See <a
        href="https://dev.mailjet.com/email/guides/#authentication">here</a>.
```

Please add title to your parameters.

File: airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/spec.yaml [hunk 10]
```diff
@@ -0,0 +1,21 @@
+documentationUrl: https://dev.mailjet.com/email/guides/
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-mail/source_mailjet_mail/spec.yaml:1]:
```suggestion
documentationUrl: https://docs.airbyte.com/integrations/sources/mailjet-mail
```

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
A few comments and it is ready to go! Thanks for the contribution @haithem-souala

**Review by marcosmarxm — APPROVED** [review]
Thanks @haithem-souala I'll publish the connector later.

**Comment by RealChrisSean** [comment]
Hi @haithem-souala is [haithem.souala@woopit.fr](mailto:haithem.souala@woopit.fr) the best email to contact you about payouts?

**Pull request merged**